x-pack/plugins/enterprise_search/server/routes/app_search/engines.test.ts

```diff
@@ -108,6 +108,73 @@ describe('engine routes', () => {
     });
   });
 
+  describe('POST /api/app_search/engines', () => {
+    const AUTH_HEADER = 'Basic 123';
+    const mockRequest = {
+      headers: {
+        authorization: AUTH_HEADER,
+      },
+      body: {
+        name: 'some-engine',
+        language: 'en',
+      },
+    };
+
+    let mockRouter: MockRouter;
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+      mockRouter = new MockRouter({
+        method: 'post',
+        path: '/api/app_search/engines',
+        payload: 'body',
+      });
+
+      registerEnginesRoutes({
+        ...mockDependencies,
+        router: mockRouter.router,
+      });
+    });
+
+    it('creates a request handler', () => {
+      mockRouter.callRoute(mockRequest);
+
+      expect(mockRequestHandler.createRequest).toHaveBeenCalledWith({
+        path: '/as/engines',
+        body: {
+          name: 'some-engine',
+          language: 'en',
+        },
+      });
+    });
```

Comment:
Follow up to above comment, we can simplify this to

```suggestion
    it('creates a request handler', () => {
      expect(mockRequestHandler.createRequest).toHaveBeenCalledWith({
        path: '/as/engines',
      });
    });
```

Reply:
Hmmm maybe I'm confused about your suggestion.  If I take our the `mockRouter.callRoute` call, this won't call `mockRequestHandler.createRequest` at all.  If I do restore that, and take out the `body` though I get the following error in `jest`

```
  ● engine routes › POST /api/app_search/engines › creates a request handler

    expect(jest.fn()).toHaveBeenCalledWith(...expected)

    - Expected
    + Received

      Object {
    +   "body": Object {
    +     "language": "en",
    +     "name": "some-engine",
    +   },
        "path": "/as/engines",
      },

    Number of calls: 1

      129 |       mockRouter.callRoute({ body: { name: 'some-engine', language: 'en' } });
      130 | 
    > 131 |       expect(mockRequestHandler.createRequest).toHaveBeenCalledWith({
          |                                                ^
      132 |         path: '/as/engines',
      133 |       });
      134 |     });
```

Reply:
Hm I might have to jump on a screenshare with you to figure this out, but the test should pass as-is once the changes in [this comment](https://github.com/elastic/kibana/pull/88716#discussion_r560453959) are in. I just did this like 30+ times for Workplace Search's routes so I'm fairly confident that should be the case - you no longer need to call `mockRouter.callRoute()` once `enterpriseSearchRequestHandler.createRequest()` is no longer nested in an async fn

Reply:
oh yeah looking at your code it's because you still have the async wrapper which isn't necessarily if you don't need access to anything in `request` to pass custom stuff. Just do

```
    enterpriseSearchRequestHandler.createRequest({
      path: '/as/engines',
    })
```

and the test will pass